SCons/Script/Main.xml

```diff
@@ -943,6 +944,51 @@ Example:
 SetOption('max_drift', 0)
 </example_commands>
 </summary>
+
+
 </scons_function>
 
+    <scons_function name="ValidateOptions">
+        <arguments signature="global">
+            ([throw_exception=False])
+        </arguments>
+
+        <summary>
+            <para>
+                Check that all the options specified on the command line are either defined by SCons itself
+                or defined by &f-link-AddOption;.
+            </para>
+            <para>
+                If <parameter>throw_exception</parameter> is True, &f-ValidateOptions; will raise a
+                <literal>SConsBadOptionError</literal>
+                exception instead of printing an error message and exiting with an error status.
+            </para>
+            <para>
+                If there are any command line options not defined.
+                Calling this function will cause SCons to issue an error message and then exit with an error exit
+                status.
+            </para>
+            <para>
+                This function should only be called after the last &f-link-AddOption; call in your &SConscript;
+                logic.
+            </para>
+            <para>
+                Be aware that some tools call &f-link-AddOption;, if you are getting error messages for arguments
+                that they add, you will need to ensure that you load those tools before you call &f-ValidateOptions;.
+            </para>
+
+            <para>
+                Example:
+            </para>
+
+<example_commands>
```

Comment:
looks like an incomplete example - add the `try` and the function call!

Reply:
good catch.. I half pasted the test cases logic here. Should be addressed in the next push.

Reply:
ok, see it now.